sdk/src/main/java/com/bugsnag/android/User.java

```diff
@@ -36,27 +31,131 @@ public void toStream(@NonNull JsonStream writer) throws IOException {
         writer.endObject();
     }
 
+    @Nullable
     public String getId() {
         return id;
     }
 
-    public void setId(String id) {
-        this.id = id;
-    }
-
+    @Nullable
     public String getEmail() {
         return email;
     }
 
-    public void setEmail(String email) {
-        this.email = email;
-    }
-
+    @Nullable
     public String getName() {
         return name;
     }
 
-    public void setName(String name) {
-        this.name = name;
+    /**
+     * Use this to create a new object holding user related details
+     *
+     * @return a builder for {@link User}
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Create a new user object based on a previous user object.
+     *
+     * @param user the previous user object.
+     */
+    public static Builder builder(User user) {
+        return new Builder(user);
+    }
```

Comment:
I think we can get rid of the two methods above, and favour one way of creating a builder:

`User user = new User.Builder().id("Foo").build();`

The second method could be replaced by something like the following:

`User.Builder builder = user.toBuilder();`

Reply:
Good idea, this would also simplify internal use as `user.toBuilder()...` is nicer than `User.builder(user)...`.

Reply:
I will also drop the `User.builder()` method too to save another method as the builder constructors are public anyways.